packages/slate-editor/src/modules/editor-v4-story-bookmark/components/StoryBookmarkElement/StoryBookmarkBlock.tsx

```diff
@@ -0,0 +1,65 @@
+import type { Story } from '@prezly/sdk';
+import type { StoryBookmarkNode } from '@prezly/slate-types';
+import { StoryBookmarkLayout } from '@prezly/slate-types';
+import React, { useRef, useState } from 'react';
+
+import { useResizeObserver, utils } from '#lib';
+
+import { BookmarkCard } from '#modules/editor-v4-components';
+
+interface StoryBookmarkBlockProps {
+    isSelected: boolean;
+    story: Story;
+    element: StoryBookmarkNode;
+}
+
+const HORIZONTAL_LAYOUT_MIN_WIDTH = 480;
+
+export function StoryBookmarkBlock({ story, element, isSelected }: StoryBookmarkBlockProps) {
+    const card = useRef<HTMLDivElement | null>(null);
+    const [isSmallViewport, setSmallViewport] = useState(false);
+
+    const showThumbnail = element.show_thumbnail && story.oembed.thumbnail_url;
+
+    const isEmpty =
+        !showThumbnail &&
+        utils.isEmptyText(story.oembed.title) &&
+        utils.isEmptyText(story.oembed.description);
+
+    const actualLayout = !showThumbnail
+        ? StoryBookmarkLayout.HORIZONTAL
+        : isSmallViewport
+        ? StoryBookmarkLayout.VERTICAL
+        : element.layout;
```

Comment:
Maybe you should consider something like that? `smallOrHorizontalViewport` is just a fast idea, if you have a better name for that feel free to change :)

```suggestion
    const smallOrHorizontalViewport = !showThumbnail
        ? StoryBookmarkLayout.HORIZONTAL
        : isSmallViewport;
    const actualLayout = smallOrHorizontalViewport
        ? StoryBookmarkLayout.VERTICAL
        : element.layout;
```